parsl/dataflow/dflow.py

```diff
@@ -796,15 +795,26 @@ def add_executors(self, executors):
             if hasattr(executor, 'provider'):
                 if hasattr(executor.provider, 'script_dir'):
                     executor.provider.script_dir = os.path.join(self.run_dir, 'submit_scripts')
-                    if executor.provider.channel.script_dir is None:
-                        executor.provider.channel.script_dir = os.path.join(self.run_dir, 'submit_scripts')
-                        if not executor.provider.channel.isdir(self.run_dir):
-                            parent, child = pathlib.Path(self.run_dir).parts[-2:]
-                            remote_run_dir = os.path.join(parent, child)
-                            executor.provider.channel.script_dir = os.path.join(remote_run_dir, 'remote_submit_scripts')
-                            executor.provider.script_dir = os.path.join(self.run_dir, 'local_submit_scripts')
-                    executor.provider.channel.makedirs(executor.provider.channel.script_dir, exist_ok=True)
                     os.makedirs(executor.provider.script_dir, exist_ok=True)
+
+                    if hasattr(executor.provider, 'channels'):
+                        logger.debug("Creating script_dir across ad-hoc cluster")
```

Comment:
This log shouldn't refer to ad-hoc clusters unless this code lives in the ad-hoc provider - maybe more like "creating script_dir across multiple channels" and "creating script_dir on one channel" in the else clause